### PR TITLE
Zap 27 UI client core

### PR DESCRIPTION
**Pull request by Valegox**
## Describe your changes

Add API class that connects to the server and handles read / write buffers.

## Issue ticket number and link

https://uwuclub.atlassian.net/browse/ZAP-27?atlOrigin=eyJpIjoiN2ZkZTNmMjI4MTQ3NGNiNDg5MmU0ODYyNjc2M2ZmMzIiLCJwIjoiaiJ9

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [ ] Do we need to implement unit tests?

### CODE REVIEW

**Review by Beafowl-Pull — APPROVED** [review]

File: src/gui/cli/ParserData.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    unsigned int ParserData::getPort() const
+    {
+        return _port;
+    }
+
+    std::string ParserData::getMachineName() const
+    {
+        return _machineName;
```
Comment (GlassAlo) [inline src/gui/cli/ParserData.cpp:64]: const ref

File: src/gui/cli/ParserData.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        if (_port == 0) {
+            throw ParserException("Invalid port given");
+        }
+        if (_machineName.empty()) {
+            _machineName = "localhost";
+        }
+    }
+
+    std::string ParserData::getAddress() const
```
Comment (GlassAlo) [inline src/gui/cli/ParserData.cpp:52]: const ref

File: src/gui/cli/ParserData.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#include "ParserData.hpp"
+
+namespace Zappy::GUI {
+    ParserData::ParserData(std::string aAddress, unsigned int aPort, std::string aMachineName)
```
Comment (GlassAlo) [inline src/gui/cli/ParserData.cpp:4]: soit const ref pour les str

soit std::move
ParserData::ParserData(std::string aAddress, unsigned int aPort, std::string aMachineName) : 
     _address(std::move(aAddress))

File: src/gui/cli/ServerData.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    SOUTH,
+    WEST
+};
+
+struct ServerData
+{
+        std::pair<unsigned int, unsigned int> _mapSize;
+        std::vector<std::vector<std::string>> _mapTiles;
+        std::vector<std::string> _teamNames;
+        std::pair<unsigned int, unsigned int> _playerPosition;
```
Comment (GlassAlo) [inline src/gui/cli/ServerData.hpp:17]: une class player
remplacer par vector de cette class

File: src/gui/main.cpp [hunk 5]
```diff
@@ -3,12 +3,23 @@
 //
 
 #include <iostream>
+#include "ClientApi.hpp"
+#include "ParserData.hpp"
 
 int main(__attribute__((unused)) int argc, __attribute__((unused)) char **argv)
```
Comment (GlassAlo) [inline src/gui/main.cpp:9]: [[maybe_unused]]

**Review by GlassAlo — CHANGES_REQUESTED** [review]

File: src/gui/cli/ParserData.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        _address = aAddress;
+        _port = aPort;
+        _clientName = aClientName;
+    }
+
+    void ParserData::parseData(int argc, char **argv)
+    {
+        std::string myFirstOpt;
+        std::string mySecondOpt;
+        int opt = 0;
```
Comment (GlassAlo) [inline src/gui/cli/ParserData.cpp:16]: var nommage convention

File: src/gui/cli/ParserData.hpp [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        public:
+            ParserData(const std::string &aAddress, unsigned int aPort, const std::string &aClientName);
+            ~ParserData() = default;
+
+            /**
+             * @brief update the address, port and machine name
+             * @param argv
+             * @param argc
+             */
+            void parseData(int argc, char **argv);
```
Comment (GlassAlo) [inline src/gui/cli/ParserData.hpp:17]: var nommage convention

File: src/gui/cli/Player.hpp [hunk 8]
```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <array>
+#include <iostream>
+#include "ServerData.hpp"
+
+#define INVENTORY_SIZE 7
```
Comment (GlassAlo) [inline src/gui/cli/Player.hpp:7]: replace with
const constexpr int INVENTORY_SIZE = 7
or
const constexpr uin8_t INVENTORY_SIZE = 7
(not sure about the uint8_t, find the good one)

**Review by GlassAlo — CHANGES_REQUESTED** [review]

**Review by GlassAlo — CHANGES_REQUESTED** [review]
Gimme the zappy_ai back

**Review by GlassAlo — APPROVED** [review]

**Pull request merged**